src/lib/operators/table_scan.cpp

```diff
@@ -177,14 +181,28 @@ void TableScan::_init_scan() {
     return;
   }
 
-  if (is_variant(parameter)) {
+  if (condition == PredicateCondition::Between) {
```

Comment:
The following if constructs in this function seem a bit inconsistent with the ones above.

Reply:
What do you suggest? Those are the conditions. :(

Reply:
I just thought since the first two blocks (Like and IsNull) return, we could write the whole thing as if .. else if .. else if .. else .. construct or if .. return if .. return

Reply:
I believe that was done on purpose so that it looks more like a list of different cases (similar to a switch) and less like guard clauses.